packages/plugin-session-replay-browser/src/session-replay.ts

```diff
@@ -338,10 +386,26 @@ class SessionReplay implements SessionReplayEnrichmentPlugin {
     }
   }
 
+  async storeShouldRecordForSession(sessionId: number, shouldRecord: boolean) {
```

Comment:
Can you share a little context on why we need to store the should record for session? Is it for multiple tabs or for the nexts session load?

Reply:
Yeah its for consistency across multiple tabs or a page refresh. We don't want to recalculate whether a session should record when a page refreshes, otherwise we might decide to record part of a session only, that would result in some really confusing recordings when part of a session is deemed included in a sample and another part is not. This allows for consistent behavior for an entire session, no matter how many page loads there are